wp-includes/components/class-wp-service-worker-caching-routes-component.php

```diff
@@ -156,7 +165,7 @@ public function get_script() {
 	 * @param array $matches Matches.
 	 * @return string Replaced string.
 	 */
-	protected function convert_snake_case_to_camel_case_callback( $matches ) {
+	protected static function convert_snake_case_to_camel_case_callback( $matches ) {
```

Comment:
Actually, I think I neglected to make it `public` instead of `protected`. I made this change for the sake of PHP 5.3 which isn't able to use a protected class method in a `preg_replace_callback` call, if I recall correctly.

Reply:
I'm fairly certain it does support protected/private callbacks, even in 5.2. The only things that don't are WP hooks :)

Reply:
See http://php.net/manual/de/function.preg-replace-callback.php#110353

Reply:
Ah, I remember the reason. The `protected` issue is what I had about the `replace_exported_variable` method actually.

The actual reason why I made this method `static` was because it is called from `prepare_strategy_args_for_js_export` which is a static method. This method is called in `WP_Service_Worker_Caching_Routes_Component::get_script()` and in `\WP_Service_Worker_Navigation_Routing_Component::serve()`.

Reply:
I'd prefer to not rely on static methods unless absolutely necessary. For this specific case, to me the `prepare_strategy_args_for_js_export()` appears to be a utility method. If it was just used in one class, it could remain non-static and protected/private. Since it is however needed by another class too, I suggest to either outsource it as a static method into a utility class, or just as a plan utility function. Having it in `WP_Service_Worker_Caching_Routes_Component` although it is also needed by `WP_Service_Worker_Navigation_Routing_Component` feels confusing.

Reply:
I realized that this method should have been put into the `WP_Service_Worker_Caching_Routes` class.

Reply:
Ah! This makes a lot more sense to me, and then having the method being static for outside access appears somewhat okay too - I'm sure we can find a better architecture for this, but I'm okay with proceeding as is for now.